ceph-releases/luminous/ubuntu/16.04/daemon/zap_device.sh

```diff
@@ -11,70 +105,34 @@ function zap_device {
   fi
 
   if [[ "${OSD_DEVICE}" == "all_ceph_disks" ]]; then
-    for type in " data" " journal" " block" " block.wal" " block.db"; do
-      for disk in $(blkid -t PARTLABEL="ceph$type" -o device | uniq); do
-        dev="$dev ${disk%?}"
-      done
-    done
-    # get a uniq list of devices to wipe
-    disks=$(echo "$dev" | tr ' ' '\n' | nl | sort -u -k2 | sort -n | cut -f2-)
-    ceph-disk zap "$disks"
-  else
-    # testing all the devices first so we just don't do anything if one device is wrong
-    for device in $(comma_to_space "${OSD_DEVICE}"); do
-      if [[ $(stat --format=%F "$device" 2> /dev/null) != "block special file" ]]; then
-        log "Provided device $device does not exist."
-        exit 1
-      fi
-      # if the disk passed is a raw device AND the boot system disk
-      if echo "$device" | grep -sqE "${device_match_string}" && parted -s "$(echo "$device" | grep -Eo "${device_match_string}")" print | grep -sq boot; then
-        log "Looks like $device has a boot partition,"
-        log "if you want to delete specific partitions point to the partition instead of the raw device"
-        log "Do not use your system disk!"
-        exit 1
-      fi
-    done
+    OSD_DEVICE=$(get_all_ceph_devices);
+  fi
+  # testing all the devices first so we just don't do anything if one device is wrong
+  for device in $(comma_to_space "${OSD_DEVICE}"); do
+    # Check if ${device} is well a block device
+    [[ -b "${device}" ]] || log "Provided device ${device} does not exist or is not a valid block device."
 
-    # look for Ceph encrypted partitions
-    local ceph_dm
-    ceph_dm=$(blkid -t TYPE="crypto_LUKS" "${OSD_DEVICE}"* -o value -s PARTUUID || true)
+    partitions=$(get_child_partitions "${device}")
+    # if the disk passed is a raw device AND the boot system disk
+    [[ $(lsblk --nodeps -no LABEL "${1}") == "boot" ]] && log "Looks like ${device} has a boot partition," \
```

Comment:
This won't work.
You need to keep adding `&&` the end of each line and leave the last `exit 1`.

Reply:
ok, was not sure for this.